src/generator.cs

```diff
@@ -3393,6 +3399,320 @@ bool FilterMinimumVersion (AvailabilityBaseAttribute aa)
+		// Attributes on the _target_ context, the class itself or the target of the protocol inlining
+		List<AvailabilityBaseAttribute> parentContextAvailability = GetAllParentAttributes (context);
+		// (Optional) Attributes from the inlined protocol type itself
+		List<AvailabilityBaseAttribute> inlinedTypeAvailability = inlinedType != null ? GetAllParentAttributes (inlinedType) : null;
+
+		// We must consider attributes if we have any on our type, or if we're inlining and that inlined type has attributes
+		// If neither are true, we have zero attributes that are relevant
+		bool shouldConsiderAttributes = memberAvailability.Any () || inlinedTypeAvailability != null && inlinedTypeAvailability.Any ();
+		if (shouldConsiderAttributes) {
+			// We will consider any inlinedType attributes first, if any, before any from our parent context
+			List<AvailabilityBaseAttribute> availabilityToConsider = new List<AvailabilityBaseAttribute>();
+			if (inlinedTypeAvailability != null) {
+				availabilityToConsider.AddRange (inlinedTypeAvailability);
+			}
+			availabilityToConsider.AddRange (parentContextAvailability);
+
+			// We do not support Watch, so strip from both our input sources before any processing
+			memberAvailability = memberAvailability.Where (x => x.Platform != PlatformName.WatchOS).ToList();
+			availabilityToConsider = availabilityToConsider.Where (x => x.Platform != PlatformName.WatchOS).ToList();
+
+			// Add any implied non-catalyst introduced (Catalyst will come later)
+			availabilityToConsider.AddRange (AddUnlistedAvailability (context, availabilityToConsider));
+
+			// Copy down any unavailable from the parent before expanding, since a [NoMacCatalyst] on the type trumps [iOS] on a member
+			CopyValidAttributes (memberAvailability, availabilityToConsider.Where (attr => attr.AvailabilityKind != AvailabilityKind.Introduced).Select (attr => CloneFromOtherPlatform (attr, attr.Platform)));
+
+			// Add implied catalyst from [iOS] _before_ copying down from parent if no catalyst attributes
+			// As those take precedent. We will do this a second time later in a moment..
+			AddImpliedCatalyst (memberAvailability);
+
+			// Now copy it down introduced from the parent
+			CopyValidAttributes (memberAvailability, availabilityToConsider.Where (attr => attr.AvailabilityKind == AvailabilityKind.Introduced).Select (attr => CloneFromOtherPlatform (attr, attr.Platform)));
+
+			// Now expand the implied catalyst from [iOS] a second time
+			// This is needed in some cases where the only iOS information is in the
+			// parent context, but we want to let any local iOS override a catalyst on the parent
+			AddImpliedCatalyst (memberAvailability);
+
+			// If all of this implication gives us something silly, like being introduced
+			// on a type that is on a namespace we don't support, ignore those Supported
+			StripIntroducedOnNamespaceNotIncluded (memberAvailability, context);
+			if (inlinedType != null) {
```

Comment:
nit: just a few places in this method that could use "is not null" instead of "!= null"
but very small thing in PR of this magnitude!